vcs_repo.go

```diff
@@ -0,0 +1,300 @@
+package gps
+
+import (
+	"bytes"
+	"encoding/xml"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/vcs"
+)
+
+type gitRepo struct {
+	*vcs.GitRepo
+}
+
+func (r *gitRepo) Get() error {
+	out, err := runFromCwd("git", "clone", "--recursive", r.Remote(), r.LocalPath())
+
+	// There are some windows cases where Git cannot create the parent directory,
+	// if it does not already exist, to the location it's trying to create the
+	// repo. Catch that error and try to handle it.
+	if err != nil && r.isUnableToCreateDir(err) {
+		basePath := filepath.Dir(filepath.FromSlash(r.LocalPath()))
+		if _, err := os.Stat(basePath); os.IsNotExist(err) {
+			err = os.MkdirAll(basePath, 0755)
+			if err != nil {
+				return vcs.NewLocalError("Unable to create directory", err, "")
+			}
+
+			out, err = runFromCwd("git", "clone", r.Remote(), r.LocalPath())
+			if err != nil {
+				return vcs.NewRemoteError("Unable to get repository", err, string(out))
```

Comment:
I realize this is mostly just copying over from Masterminds/vcs, but let's improve all these errors in two ways:

1. Per standards, don't capitalize the first letter
2. In methods where the same text is used over and over, differentiate the messages. (The repetition might be fine in this method, but there's other places where it potentially obscures what's really going on)